### PR TITLE
21.1.6+

### PR DESCRIPTION
**Pull request by kaatula**

### CODE REVIEW

File: Readme.md [hunk 1]
```diff
@@ -3,22 +3,14 @@
...
 
 The [Extract Data Source](https://docs.devexpress.com/Dashboard/115900) improves performance when a complex query or a stored procedure takes a significant time to get data from a database. 
 
-The [DashboardExtractDataSource](https://docs.devexpress.com/Dashboard/DevExpress.DashboardCommon.DashboardExtractDataSource) class implements the Extract Data Source concept and allows you to request the data once and save it in the compressed and optimized form to a file. Subsequently tan application can retrieve data from that file or create a new file when data is updated. 
-
-This concept is described in the <a href="https://community.devexpress.com/blogs/news/archive/2016/08/16/faster-dashboards-with-the-data-extract-source.aspx">Faster Dashboards with the “Data Extract” Source</a> blog.
+The [DashboardExtractDataSource](https://docs.devexpress.com/Dashboard/DevExpress.DashboardCommon.DashboardExtractDataSource) class implements the Extract Data Source concept and allows you to request the data once and save it in the compressed and optimized form to a file. Subsequently tan application can retrieve data from that file or create a new file when data is updated.
```
Comment (natakazakova) [inline Readme.md]:
```suggestion
The [DashboardExtractDataSource](https://docs.devexpress.com/Dashboard/DevExpress.DashboardCommon.DashboardExtractDataSource) class implements the Extract Data Source concept and allows you to request the data once and save it in the compressed and optimized form to a file. Subsequently an application can retrieve data from that file or create a new file when data is updated.
```

File: Readme.md [hunk 2]
```diff
@@ -3,22 +3,14 @@
...
-
-* [Default.aspx](./CS/ASP_WebDashboard/Default.aspx) (VB: [Default.aspx](./VB/ASP_WebDashboard/Default.aspx))
-* [Default.aspx.cs](./CS/ASP_WebDashboard/Default.aspx.cs) (VB: [Default.aspx.vb](./VB/ASP_WebDashboard/Default.aspx.vb))
-<!-- default file list end -->
 
 # Dashboard for Web Forms - How to Use DashboardExtractDataSource to provide data
 
 This example demonstrates basic approaches and code snippets that can be used to bind the [ASPxDashboard](https://docs.devexpress.com/Dashboard/DevExpress.DashboardWeb.ASPxDashboard) control to the [DashboardExtractDataSource](https://docs.devexpress.com/Dashboard/DevExpress.DashboardCommon.DashboardExtractDataSource). 
 
 The [Extract Data Source](https://docs.devexpress.com/Dashboard/115900) improves performance when a complex query or a stored procedure takes a significant time to get data from a database.
```
Comment (arminal) [inline Readme.md]:
```suggestion
The [Extract Data Source](https://docs.devexpress.com/Dashboard/115900) improves performance when a complex query or a stored procedure takes a significant amount of time to get data from a database. 
```

File: Readme.md [hunk 3]
```diff
@@ -3,22 +3,14 @@
...
 
 The [Extract Data Source](https://docs.devexpress.com/Dashboard/115900) improves performance when a complex query or a stored procedure takes a significant time to get data from a database. 
 
-The [DashboardExtractDataSource](https://docs.devexpress.com/Dashboard/DevExpress.DashboardCommon.DashboardExtractDataSource) class implements the Extract Data Source concept and allows you to request the data once and save it in the compressed and optimized form to a file. Subsequently tan application can retrieve data from that file or create a new file when data is updated. 
-
-This concept is described in the <a href="https://community.devexpress.com/blogs/news/archive/2016/08/16/faster-dashboards-with-the-data-extract-source.aspx">Faster Dashboards with the “Data Extract” Source</a> blog.
+The [DashboardExtractDataSource](https://docs.devexpress.com/Dashboard/DevExpress.DashboardCommon.DashboardExtractDataSource) class implements the Extract Data Source concept and allows you to request the data once and save it in the compressed and optimized form to a file. Subsequently an application can retrieve data from that file or create a new file when data is updated.
```
Comment (arminal) [inline Readme.md]:
```suggestion
The [DashboardExtractDataSource](https://docs.devexpress.com/Dashboard/DevExpress.DashboardCommon.DashboardExtractDataSource) class implements the Extract Data Source concept, and allows you to request data once and save it in the compressed and optimized form to a file. Subsequently, an application can retrieve data from that file or create a new file when data is updated.
```

File: Readme.md [hunk 4]
```diff
@@ -69,35 +60,51 @@ if (!File.Exists(extractFileName)) {
...
+	    sb.AppendLine($"{DateTime.Now.ToString("T")} - File Updated - {result} - {Path.GetFileName(fileName)}. ");
+	});
+    // Wait until the data is refreshed in the Extract Data Source.
+    task.Wait();
+    return sb.ToString();
 }
 ```
 
-In this example, data is extracted on a button click. However, in a real-life application, this solution can be insufficient (e.g. the site may be deployed to the web farm server). We recommend creating a separate windows service that should update data automatically every hour or every day.
+In this example, click a button to extract data. However, in a real-life application, this solution can be insufficient (for example, the site may be deployed to the web farm server). We recommend that you create a separate windows service that should update data automatically every hour or every day.
```
Comment (arminal) [inline Readme.md]:
```suggestion
In this example, click a button to extract data. However, in a real-life application, this solution can be insufficient (for example, the site may be deployed to the web farm server). We recommend that you create a separate windows service that updates data automatically every hour or every day.
```

**Comment by arminal** [comment]
#GrammarReview

Please take a look at article usage and review phrasing.

File: Readme.md [hunk 5]
```diff
@@ -69,35 +60,51 @@ if (!File.Exists(extractFileName)) {
...
+To update the data extract file and load the updated data in ASPxDashboard, send an AJAX request to the server and call the [DashboardExtractDataSource.UpdateFile](https://docs.devexpress.com/Dashboard/DevExpress.DashboardCommon.DashboardExtractDataSource.UpdateFile(DashboardExtractDataSource--Action-String--ExtractUpdateResult---Action-String--ExtractUpdateResult-)) method there. Once a new file is created on the server, you can return the callback back to the client and call the [ASPxClientDashboard.ReloadData](http://docs.devexpress.devx/Dashboard/js-ASPxClientDashboard?p=netframework#js_aspxclientdashboard_reloaddata) method to reload the control with new data:
```
Comment (natakazakova) [inline Readme.md]:
```suggestion
To update the data extract file and load the updated data in ASPxDashboard, send an AJAX request to the server and call the [DashboardExtractDataSource.UpdateFile](https://docs.devexpress.com/Dashboard/DevExpress.DashboardCommon.DashboardExtractDataSource.UpdateFile(DashboardExtractDataSource--Action-String--ExtractUpdateResult---Action-String--ExtractUpdateResult-)) method there. Once a new file is created on the server, you can return the callback back to the client and call the [ASPxClientDashboard.ReloadData](https://docs.devexpress.com/Dashboard/js-ASPxClientDashboard?p=netframework#js_aspxclientdashboard_reloaddata) method to reload the control with new data:
```

**Pull request merged**